### DISCUSSION TITLE
Potential Bug: no reinitialisation of the vector database

### DISCUSSION THREAD

**Discussion by flefevre** [open-webui/open-webui#6185] [General]
# Bug Report

## Installation Method

I used the docker installation

## Environment

- **Open WebUI Version:** v0.3.29
- ** Ollama** : 0.3.2
- **Chroma**: 0.5.5
- **Operating System:** Ubuntu 24.04

**Confirmation:**

- [X ] I have read and followed all the instructions provided in the README.md.
- [X] I am on the latest version of both Open WebUI and Ollama.
- [ X] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

When I click on the "Reset Vectorization Database" option in the admin interface and then open a new discussion referencing a collection or document with a #, the system should recognize and vectorize the documents correctly.

## Actual Behavior:

After resetting the vectorization database and opening a new discussion referencing a collection or document with a #, the system responds with "I don't know what this collection is talking about," as if the documents have not been vectorized.

## Description

**Bug Summary:**
The system fails to recognize vectorized documents after resetting the vectorization database.

## Reproduction Details

**Steps to Reproduce:**
1. Navigate to the admin interface.
2. Click on the "Reset Vectorization Database" option.
3. Confirm the reset action.
4. Open a new discussion and reference a collection or document using a #.
5. Observe the system response indicating that it does not recognize the collection.

**Comment by tjbck** [comment]
This is intended as If you reset the vector db, it will remove all documents and it's embeddings from the vector db. 

With that being said, It has been refactored in dev to also remove knowledge base simultaneously 
![image](https://github.com/user-attachments/assets/53592834-ef29-4c94-acec-9543bd9d1f8c)

**Reply by Hsun1128** [reply]
The knowledge base should not be deleted after resetting the vector database, because I want to regenerate the vector database after modifying the Embedding Model, Chunk Size and Chunk Overlap. However, the option to delete the knowledge base might need to be updated to include deleting the vector database to avoid potential space consumption.

**Reply by dnesting** [reply]
I just hit this as well.  When I read "reset" I interpret that as "to set again, or set differently", not "delete".  I expected a new vector database would be created with updated settings for the existing collections.  It doesn't make sense to delete the vector database if the collection becomes permanently unusable, but maybe that's why the last screenshot shows them combined.

There should be some way to regenerate the vector database without forcing us to delete and re-upload collections.

**Reply by tjbck** [reply]
Agreed 👍 PR welcome!
